### ISSUE TITLE
Windows Instaler don't work with 32 bit Windows

### ISSUE THREAD

**Issue by filchef** [NewEconomyMovement/NemCommunityClient#280]
My and many more weak computers running Windows 32 bit and for common user is problem to instal 64 bit Windows so I think Installer must recognize and work in 32 bit systems.

**Comment by gimer** [comment]
I'm not sure if we're going to change that. On 32-bit system it is suggested to run standalone package instead.

**Comment by filchef** [comment]
So for me no problem but for not so experienced users this will be problem may be two different application for 32 bit and for 64 bit Windows if is not possible to make one application for 32/64.

**Comment by gimer** [comment]
If you can upgrade - do upgrade, there's really no good reason to use 32-bit system these days.

a) 64-bit windows is more secure (for various reasons)
b) if you're running 32-bit OS, (unless you've switched to PAE and/or 4GT), applications can maximally use only 2G of ram

**Comment by fyahfox** [comment]
1) There is still no Firefox 64 bit. And there are still a lot of other older browsers in use (especially by non technical people).
2) Oracle itself suggests to use 32 bit Java, if you have a 32 bit browser (https://www.java.com/en/download/faq/java_win64bit.xml). 

It would be best if the installer could check if the OS is 32 or 64 and check the default browser as well if it is 32 or 64 bit and then suggest to install 64 bit Java or accept a 32 bit Java.

**Comment by gimer** [comment]
Since we're NOT using webstart anymore, you can safely use 32-bit browser and 64-bit java.

The only reason oracle suggest this is (as you've noted) for browser compatibility (that is running java applets to be more precise). But that is not the case for nem. So as I've said, you're safe to run 32-bit browser and run NEM using 64-bit java.

**Comment by fyahfox** [comment]
Yes, I never meant to say, your'e not safe to do that. But as I explained, there are not few people with Java 32 bit (Firefox users e.g.) and they get into trouble with the installer, because it doesn't accept 32 bit Java although that could work (it does with stand alone).

**Comment by gimer** [comment]
@fyahfox installer is supposed to be for newbies. It will really be better if they will run 64-bit version (quite a few reasons). If someone wants to run 32-bit version, he's considered as pro/advanced user, and he should use standalone

**Comment by fyahfox** [comment]
"If someone wants to run 32-bit version, he's considered as pro/advanced user"

You mean 32-bit Java, right?
If so: I don't agree, because average users install what is recommended, and oracle itself recommends 32-bit Java for 32-bit browsers (e.g. Firefox, which is always 32-bit on Windows).

**Comment by gimer** [comment]
@fyahfox when it comes to windows, 64-bit processes are more secure, for multiple reasons. And nis handles your money, we don't want to encourage people to run nis process as 32-bit process...

**Comment by fyahfox** [comment]
I see, I never knew it was a security issue.
Why don't you display that as a warning in the installer?

**Comment by gimer** [comment]
I'd like to prepare longer rationale, but that has lower priority as of now.

**Comment by fyahfox** [comment]
ok
